scipy/interpolate/_aaa.py

```diff
@@ -57,6 +57,12 @@ class AAA:
     max_terms : int, optional
         Maximum number of terms in the barycentric representation, defaults to ``100``.
         Must be greater than or equal to one.
+    cleanup : bool, optional
+        Automatic removal of Froissart doublets, defaults to ``True``.
```

Comment:
1. Could you briefly parse what Froissart doublets are, here or in the Notes section. 
I don't think the term is that widely known.
To make sure I understood the original paper right: these are pairs of poles with vanishingly small residues of differing signs close to each other?

2. Hopefully there is a discussion or at least a mention somewhere that the algorithm is going to generate these if `max_terms` is too large (if this is indeed correct)

3. Not sure `cleanup` the best term here.

Reply:
> 1. Could you briefly parse what Froissart doublets are, here or in the Notes section.
>    I don't think the term is that widely known.
>    To make sure I understood the original paper right: these are pairs of poles with vanishingly small residues of differing signs close to each other?

Almost, it is a pole-zero pair where the pole has a tiny residue. Consider approximating $f$, we can add $g(z) = \epsilon (z-z_0)/(z-z_0-\delta)$, where $\epsilon, \delta \ll 1$, to $f$ and its value remains basically unchanged apart from at near $z_0$ but $f+g$ has a spurious pole-zero pair.

> 2. Hopefully there is a discussion or at least a mention somewhere that the algorithm is going to generate these if `max_terms` is too large (if this is indeed correct)
Will add
> 3. Not sure `cleanup` the best term here.
Open to suggestion but this is the term they use in the paper